### ISSUE TITLE
How to hide ffmpeg's console output?

### ISSUE THREAD

**Issue by duruburak** [kkroening/ffmpeg-python#729]
Hi, I'm using pytube and ffmpeg-python.

When I try to merge audio and video that comes from pytube, ffmpeg pops up a console. Is there any way to hide it?
Thanks in advance.

I use this:
`ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)`

**Comment by hkiragi** [comment]
Add "from subprocess import CREATE_NO_WINDOW" and use "creationflags=CREATE_NO_WINDOW" for Popen. Below is updated part of "_run.py" code from ffmpeg-python library, that worked for me.

```
from subprocess import CREATE_NO_WINDOW

@output_operator()
def run_async(
    stream_spec,
    cmd='ffmpeg',
    pipe_stdin=False,
    pipe_stdout=False,
    pipe_stderr=False,
    quiet=False,
    overwrite_output=False,
):

    args = compile(stream_spec, cmd, overwrite_output=overwrite_output)
    stdin_stream = subprocess.PIPE if pipe_stdin else None
    stdout_stream = subprocess.PIPE if pipe_stdout or quiet else None
    stderr_stream = subprocess.PIPE if pipe_stderr or quiet else None
    return subprocess.Popen(
        args, stdin=subprocess.PIPE, stdout=stdout_stream, stderr=stderr_stream, creationflags=CREATE_NO_WINDOW
    )
```

**Comment by duruburak** [comment]
> 

It's the ffmpeg.exe from https://ffmpeg.org/ that pops up, not ffmpeg-python. ffmpeg-python is for bridging between python and ffmpeg. `ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)` calls ffmpeg.exe to process video. So it didn't work for me, still pops up a console window.

**Comment by hkiragi** [comment]
> > 
> 
> It's the ffmpeg.exe from https://ffmpeg.org/ that pops up, not ffmpeg-python. ffmpeg-python is for bridging between python and ffmpeg. `ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)` calls ffmpeg.exe to process video. So it didn't work for me, still pops up a console window.

Of course, it is ffmpeg.exe running at the end. Is it possible for you to share related part of your code, so I could comment better.

**Comment by duruburak** [comment]
> > > 
> > 
> > 
> > It's the ffmpeg.exe from https://ffmpeg.org/ that pops up, not ffmpeg-python. ffmpeg-python is for bridging between python and ffmpeg. `ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)` calls ffmpeg.exe to process video. So it didn't work for me, still pops up a console window.
> 
> Of course, it is ffmpeg.exe running at the end. Is it possible for you to share related part of your code, so I could comment better.

Oh sure. It's actually an .exe file I'm testing. The software still in-development, so I'll compile it again soon. I guess it should be like this:
```python
yt.streams.filter(adaptive=True, res="1080p").order_by('bitrate').desc().first().download(output_path=<path_video_part>, filename=<video_part_name>)
video_part = ffmpeg.input(<path_video_part>)
yt.streams.filter(only_audio=True).order_by('abr').desc().first().download(output_path=<path_audio_part>, filename=<audio_part_name>)
audio_part = ffmpeg.input(<path_audio_part>)
ffmpeg.output(audio_part, video_part, <path>).run(overwrite_output=True)
```
Let me know if you need anything else, thanks.

**Comment by hkiragi** [comment]
> > > > 
> > > 
> > > 
> > > It's the ffmpeg.exe from https://ffmpeg.org/ that pops up, not ffmpeg-python. ffmpeg-python is for bridging between python and ffmpeg. `ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)` calls ffmpeg.exe to process video. So it didn't work for me, still pops up a console window.
> > 
> > 
> > Of course, it is ffmpeg.exe running at the end. Is it possible for you to share related part of your code, so I could comment better.
> 
> Oh sure. It's actually an .exe file I'm testing. The software still in-development, so I'll compile it again soon. I guess it should be like this:
> 
> ```python
> yt.streams.filter(adaptive=True, res="1080p").order_by('bitrate').desc().first().download(output_path=<path_video_part>, filename=<video_part_name>)
> video_part = ffmpeg.input(<path_video_part>)
> yt.streams.filter(only_audio=True).order_by('abr').desc().first().download(output_path=<path_audio_part>, filename=<audio_part_name>)
> audio_part = ffmpeg.input(<path_audio_part>)
> ffmpeg.output(audio_part, video_part, <path>).run(overwrite_output=True)
> ```
> 
> Let me know if you need anything else, thanks.

As I understand you are trying to have a time series video presentation of a data set by using a library from yt-project.org. I have no experience with YT and it is ffmpeg compatibility. But I guess YT lets you save individual time series pictures of your data in an easier way. I may suggest you first to create your static consecutive images than use "ffmpeg-python" library to have animated video from that images. Here is the link [https://github.com/kkroening/ffmpeg-python/tree/master/examples#assemble-video-from-sequence-of-frames](url) to encode image sequences to a video. This may be an easy workaround for you.

**Comment by duruburak** [comment]
> > > > > 
> > > > 
> > > > 
> > > > It's the ffmpeg.exe from https://ffmpeg.org/ that pops up, not ffmpeg-python. ffmpeg-python is for bridging between python and ffmpeg. `ffmpeg.output(audio_part, video_part, path).run(overwrite_output=True)` calls ffmpeg.exe to process video. So it didn't work for me, still pops up a console window.
> > > 
> > > 
> > > Of course, it is ffmpeg.exe running at the end. Is it possible for you to share related part of your code, so I could comment better.
> > 
> > 
> > Oh sure. It's actually an .exe file I'm testing. The software still in-development, so I'll compile it again soon. I guess it should be like this:
> > ```python
> > yt.streams.filter(adaptive=True, res="1080p").order_by('bitrate').desc().first().download(output_path=<path_video_part>, filename=<video_part_name>)
> > video_part = ffmpeg.input(<path_video_part>)
> > yt.streams.filter(only_audio=True).order_by('abr').desc().first().download(output_path=<path_audio_part>, filename=<audio_part_name>)
> > audio_part = ffmpeg.input(<path_audio_part>)
> > ffmpeg.output(audio_part, video_part, <path>).run(overwrite_output=True)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Let me know if you need anything else, thanks.
> 
> As I understand you are trying to have a time series video presentation of a data set by using a library from yt-project.org. I have no experience with YT and it is ffmpeg compatibility. But I guess YT lets you save individual time series pictures of your data in an easier way. I may suggest you first to create your static consecutive images than use "ffmpeg-python" library to have animated video from that images. Here is the link [https://github.com/kkroening/ffmpeg-python/tree/master/examples#assemble-video-from-sequence-of-frames](url) to encode image sequences to a video. This may be an easy workaround for you.

No it's from 'pytube', a library that allows you to request streams from youtube. My bad, I didn't declare that.
```python
from pytube import YouTube
yt = YouTube("<link>")
```
